src/main/java/org/example/learningsystem/btp/destination/converter/DestinationConverter.java

```diff
@@ -0,0 +1,53 @@
+package org.example.learningsystem.btp.destination.converter;
+
+import org.example.learningsystem.btp.destination.annotation.DestinationProperty;
+import org.example.learningsystem.btp.destination.dto.DestinationDto;
+import org.example.learningsystem.btp.destination.exception.MissingNoArgsConstructorException;
+import org.example.learningsystem.btp.destination.exception.MissingPropertyException;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Field;
+
+@Component
+@Profile("cloud")
+public class DestinationConverter {
+
+    public <T> T tryConvert(DestinationDto destination, Class<T> desiredType) {
```

Comment:
tryConvert -> convert
public methods are usually called without `try` prefix, it's mostly used in private ones